httpgrpc/server.go

```diff
@@ -91,11 +113,13 @@ func HandleMethod(svr interface{}, serviceName string, desc *grpc.MethodDesc, un
 			return
 		}
 
-		dec := func(msg interface{}) error {
-			return proto.Unmarshal(req, msg.(proto.Message))
-		}
 		sts := internal.UnaryServerTransportStream{Name: fullMethod}
-		resp, err := desc.Handler(svr, grpc.NewContextWithServerTransportStream(ctx, &sts), dec, unaryInt)
+		resp, err := desc.Handler(svr, grpc.NewContextWithServerTransportStream(ctx, &sts), func(msg interface{}) error {
+			if err := unmarshal(req, msg); err != nil {
+				return status.Error(codes.InvalidArgument, err.Error())
+			}
+			return nil
```

Comment:
^^ note the new code here -- set as `InvalidArgument` if the incoming request body cannot be parsed

Reply:
I could also pull this back out into a `dec` function